src/vmm/src/vmm_config/mod.rs

```diff
@@ -97,38 +97,19 @@ impl TryInto<RateLimiter> for RateLimiterConfig {
 
 type Result<T> = std::result::Result<T, std::io::Error>;
 
-/// Structure `Writer` used for writing to a FIFO.
-pub struct Writer {
-    line_writer: io::LineWriter<File>,
+/// Create and open a FIFO for writing to it.
+/// In order to not block the instance if nobody is consuming the message that is flushed to the
+/// two pipes, we are opening it with `O_NONBLOCK` flag. In this case, writing to a pipe will
+/// start failing when reaching 64K of unconsumed content.
+fn open_fifo(fifo_path: &PathBuf) -> Result<File> {
```

Comment:
Can we make this a function of `FifoLineWriter` so that we do not have orphan functions?

Reply:
The compiler doesn't allow defining functions on `FifoLineWriter` since it's basically a `LineWriter<File>` ...

Reply:
Yap, it would need to be a struct as it was before. I don't see why the change in this commit is necessary. I preferred it as it was before. Better yet, `LineWriter` can use the [new type pattern](https://doc.rust-lang.org/1.0.0/style/features/types/newtype.html).